core/domain/topic_domain.py

```diff
@@ -562,21 +579,20 @@ def get_subtopic_index(self, subtopic_id):
                 return ind
         return None
 
-    def add_subtopic(self, subtopic_id, title):
+    def add_subtopic(self, title):
```

Comment:
I just realized that if a topic manager adds a subtopic in the frontend, then that change kinda needs to be saved immediately if they subsequently decide they want to add the subtopic card. But I can't think of a good way to handle this.

In the collection editor, when you add an exploration, the exploration gets created as a standalone entity using a random ID. Even if you later delete the exploration from the collection, the exploration still persists as an independent thing.

But what happens if you add a subtopic to the topic editor, and then discard your changes? If the subtopic model gets created as soon as you add it, then it sticks around even after changes are discarded. If it gets created only when you save the changelist, then there's no way to edit the subtopic page until you've actually committed your changes to the topic.

So ... any thoughts?

Reply:
Done, added another changelist construct similar to discussions.